src/shared/condition.c

```diff
@@ -353,6 +353,61 @@ static int condition_test_osrelease(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_dmi(Condition *c, char **env) {
+        int r;
+
+        assert(c);
+        assert(c->type == CONDITION_DMI);
+
+        for (const char *parameter = ASSERT_PTR(c->parameter);;) {
```

Comment:
a for loop that isn't looping?

Reply:
I've chosen the `condition_test_osrelease` function as template and there it is done this way. Since this is my first contribution to systemd, I thought there would be something I'm missing right now...

This should be resolved with the integration into `condition_test_firmware`.